src/realm/sync/subscriptions.cpp

```diff
@@ -971,8 +950,24 @@ int64_t SubscriptionStore::set_active_as_latest(Transaction& wt)
     sub_sets->where().greater(sub_sets->get_primary_key_column(), active.get_primary_key().get_int()).remove();
     // Mark the active set as complete even if it was previously WaitingForMark
     // as we've completed rebootstrapping before calling this.
-    active.set(m_sub_set_state, state_to_storage(SubscriptionSet::State::Complete));
-    return active.get_primary_key().get_int();
+    active.set(m_sub_set_state, state_to_storage(State::Complete));
+    auto version = active.get_primary_key().get_int();
+
+    std::list<NotificationRequest> to_finish;
+    {
+        util::CheckedLockGuard lock(m_pending_notifications_mutex);
+        splice_if(m_pending_notifications, to_finish, [&](auto& req) {
```

Comment:
why do we need this? isn't this essentially only selecting the active subscription set? we want to deliver notifications to all subscriptions removed.

Reply:
This is fulfilling notification requests for all requests for other versions and requests for the current version unless the requested state is Superseded.

Reply:
got it. I didn't pay enough attention to the second return.